extension/asapauthextension/extension.go

```diff
@@ -0,0 +1,101 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package asapauthextension // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/asapauthextension"
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	asap "bitbucket.org/atlassian/go-asap/v2"
+	"github.com/SermoDigital/jose/crypto"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configauth"
+	"google.golang.org/grpc/credentials"
+)
+
+// AsapClientAuthenticator implements ClientAuthenticator
+type AsapClientAuthenticator struct {
```

Comment:
Sorry for only catching this now, but if the software is called "ASAP", then this name here should be:

```suggestion
type ASAPClientAuthenticator struct {
```

Reply:
I've clearly been stuck in java land for too long 😆  All fixed